client/src/test/scala/it/cwmp/client/controller/ApiClientActorTest.scala

```diff
@@ -0,0 +1,245 @@
+        expectMsg(RoomsResponses.EnterPrivateSuccess)
+
+        actor ! RoomsRequests.ServiceEnterPrivate(roomID, playerAddress, playerNotificationAddress, token)
+        expectMsgType[RoomsResponses.EnterPrivateFailure]
+
+        cleanUpPrivateRoom(roomID)
+      }
+
+      "block him if entering a private room when being inside another" in {
+        val roomID = createPrivateRoomAndGetID(roomName, playersNumber)
+        val otherRoomID = createPrivateRoomAndGetID(roomName, playersNumber)
+        actor ! RoomsRequests.ServiceEnterPrivate(roomID, playerAddress, playerNotificationAddress, token)
+        expectMsg(RoomsResponses.EnterPrivateSuccess)
+
+        actor ! RoomsRequests.ServiceEnterPrivate(otherRoomID, playerAddress, playerNotificationAddress, token)
+        expectMsgType[RoomsResponses.EnterPrivateFailure]
+
+        cleanUpPrivateRoom(roomID)
+      }
+
+      "let him exit a private room" in {
+        val roomID = createPrivateRoomAndGetID(roomName, playersNumber)
+        actor ! RoomsRequests.ServiceEnterPrivate(roomID, playerAddress, playerNotificationAddress, token)
+        expectMsg(RoomsResponses.EnterPrivateSuccess)
+
+        actor ! RoomsRequests.ServiceExitPrivate(roomID, token)
+        expectMsg(RoomsResponses.ExitPrivateSuccess)
+      }
+
+      "block him from exiting a not entered room" in {
+        val fakeRoomID = "fakeRoomID"
+        actor ! RoomsRequests.ServiceExitPrivate(fakeRoomID, token)
+        expectMsgType[RoomsResponses.ExitPrivateFailure]
+      }
+
+      "let him enter a public room" in {
+        actor ! RoomsRequests.ServiceEnterPublic(playersNumber, playerAddress, playerNotificationAddress, token)
+        expectMsg(RoomsResponses.EnterPublicSuccess)
+
+        cleanUpPublicRoom(playersNumber)
+      }
+
+      "block him from entering public room twice" in {
+        actor ! RoomsRequests.ServiceEnterPublic(playersNumber, playerAddress, playerNotificationAddress, token)
+        expectMsg(RoomsResponses.EnterPublicSuccess)
+
+        actor ! RoomsRequests.ServiceEnterPublic(playersNumber, playerAddress, playerNotificationAddress, token)
+        expectMsgType[RoomsResponses.EnterPublicFailure]
+
+        cleanUpPublicRoom(playersNumber)
+      }
+
+      "let him exit a public room" in {
+        actor ! RoomsRequests.ServiceEnterPublic(playersNumber, playerAddress, playerNotificationAddress, token)
+        expectMsg(RoomsResponses.EnterPublicSuccess)
+
+        actor ! RoomsRequests.ServiceExitPublic(playersNumber, token)
+        expectMsg(RoomsResponses.ExitPublicSuccess)
+      }
+
+      "block him from exiting a non entered public room" in {
+        actor ! RoomsRequests.ServiceExitPublic(playersNumber, token)
+        expectMsgType[RoomsResponses.ExitPublicFailure]
+      }
+    }
+
+    "user not authenticated" must {
+      val badToken = "myBadToken"
+
+      "return failed messages" in {
```

Comment:
In genere la politica è di fare un test per ogni cosa, quindi uno per ogni failure, c'è una ragione per la quale li hai aggregati? @siboXD

Reply:
Si vero, li ho raggruppati un po per pigrizia, un po' perche sono già descrittivi i messaggi e si tratta dei vari fallimenti; ma si se si volesse essere perfetti sarebbe da spezzare e non ho nulla in contrario